vending-app/src/main/kotlin/com/android/vending/ExperimentAndConfigs.kt

```diff
@@ -0,0 +1,839 @@
+        @SuppressLint("HardwareIds") val builder = ExpDeviceInfo.Builder()
+        builder.androidId(
+            getString(context.contentResolver, "android_id", "")!!
+                .toLong()
+        )
+        builder.sdkInt(Build.VERSION.SDK_INT)
+        builder.buildId(Build.ID)
+        builder.buildDevice(Build.DEVICE)
+        builder.manufacturer(Build.MANUFACTURER)
+        builder.model(Build.MODEL)
+        builder.product(Build.PRODUCT)
+        builder.unknowEmpty("")
+        builder.fingerprint(Build.FINGERPRINT)
+        builder.country(Locale.getDefault().country)
+        builder.locale(Locale.getDefault().toString())
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            builder.supportAbis(Arrays.asList(*Build.SUPPORTED_ABIS))
+        }
+        return builder.build()
+    }
+
+    private fun createDeviceData(account: Account?, context: Context): DeviceData {
+        val expDeviceInfo = createExpDeviceInfo(context)
+        val expDeviceInfoWrapper = ExpDeviceInfoWrapper.Builder()
+            .unknowFieldb(4)
+            .expDeviceInfoValue(expDeviceInfo)
+            .build()
+        return DeviceData.Builder()
+            .hasAccount((if (account == null) 0 else 1).toLong())
+            .expDeviceInfoWrapperValue(expDeviceInfoWrapper)
+            .unknowFlagf(false)
+            .unknowEmptyE(ByteString.of())
+            .unknkowFieldG(DeviceDataEmptyA.Builder().build())
+            .build()
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    fun postRequest(
+        experimentRequestData: experimentRequestData,
+        context: Context?,
+        accountName: String,
+        token: String
+    ) {
+        try {
+            val url =
+                URL("https://www.googleapis.com/experimentsandconfigs/v1/getExperimentsAndConfigs" + "?r=" + experimentRequestData.actionType?.value + "&c=" + experimentRequestData.unknowFieldG)
+
+            val httpURLConnection = url.openConnection() as HttpURLConnection
+            httpURLConnection.connectTimeout = 30000
+            httpURLConnection.readTimeout = 30000
+            httpURLConnection.doOutput = true
+            httpURLConnection.instanceFollowRedirects = false
+            httpURLConnection.setRequestProperty("Accept-Encoding", null)
+            httpURLConnection.setRequestProperty("Content-Type", "application/x-protobuf")
+            httpURLConnection.setRequestProperty("Content-Encoding", "gzip")
+            httpURLConnection.setRequestProperty("Authorization", "Bearer $token")
+            httpURLConnection.setRequestProperty(
+                "User-Agent",
+                "Android-Finsky/41.2.21-31 [0] [PR] 636997666 (api=3,versionCode=84122130,sdk=31,device=redroid_arm64,hardware=redroid,product=redroid_arm64,platformVersionRelease=12,model=redroid12_arm64,buildId=SQ1D.220205.004,isWideScreen=0,supportedAbis=arm64-v8a;armeabi-v7a;armeabi) (redroid_arm64 SQ1D.220205.004); gzip"
```

Comment:
`41.2.21-31 [0] [PR] 636997666` is also defined in the manifest now. Does it need to be the same?

Lots of hardware properties seem to be hardcoded in the request. Do those matter at all and potentially change the response?